data/abi/abi_encode.go

```diff
@@ -25,23 +25,6 @@ import (
 	"strings"
 )
 
```

Comment:
@ahangsu As @algoidurovic mentioned in Slack, there's additional places where `FillBytes` can be supplied.  How do you want to go about it?

Options:
* Keep PR as revert for https://github.com/algorand/go-algorand/pull/3498/.  Open new PR for any other change.
  *  If we take the route, I'm ready to approve.
* Extend PR to catch other `FillBytes` use cases.

I don't have a strong preference and will follow your lead.

Here's the other usage:
```
~/dev/go-algorand fillbytes-enable ?28 ────────────────────────────────────────────────────────────────────────────────────────── 01:59:39 PM
❯ git --no-pager grep big.Int.FillBytes
data/transactions/logic/eval.go:// leadingZeros needs to be replaced by big.Int.FillBytes
data/transactions/logic/evalCrypto_test.go:// bitIntFillBytes is a replacement for big.Int.FillBytes from future Go
```

Additionally, I found (but didn't closely investigate) several Go upgrade placeholders for different functionality.  If we're in the neighborhood, feels prudent to address.  If you prefer, we can isolate these changes to a different PR.

```
~/dev/go-algorand fillbytes-enable ?28 ────────────────────────────────────────────────────────────────────────────────────────── 02:05:09 PM
❯ git --no-pager grep -i "go 1.15+"
data/transactions/logic/eval.go:// TODO: remove this when go-algorand is updated to go 1.15+
data/transactions/logic/eval.go:// updated to go 1.15+
data/transactions/logic/evalCrypto_test.go:// TODO: replace with elliptic.MarshalCompressed when updating to go 1.15+
```

Additionally -

Reply:
I am additionally updating the `fillBytes` and related small updates with #2852, and I decided to end here.

Reply:
@ahangsu Thanks for the additions.  Resolving.